### PR TITLE
Add background color setting to the richtext

### PR DESCRIPTION
**Pull request by c4llv07e**
Add the ability to change the background color of the text in the markup language using "[bgcolor]". This works similarly to "[color]", but it can also be null.

Highlighting words in a chat as an example:

image

### CODE REVIEW

File: Robust.Client/UserInterface/RichText/MarkupDrawingContext.cs [hunk 1]
```diff
@@ -7,26 +7,30 @@ namespace Robust.Client.UserInterface.RichText;
...
         Color = new Stack<Color>();
+        BackgroundColor = new Stack<Color?>();
         Font = new Stack<Font>();
         Tags = new List<IMarkupTag>();
     }
 
     public MarkupDrawingContext(int capacity)
     {
         Color = new Stack<Color>(capacity);
+        BackgroundColor = new Stack<Color?>(capacity);
```
Comment (metalgearsloth) [inline Robust.Client/UserInterface/RichText/MarkupDrawingContext.cs:25]: This should be nullable.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
Please use C# style guide for variable naming i.e. camelCase.

File: Robust.Client/Graphics/Font.cs [hunk 2]
```diff
@@ -131,7 +132,15 @@ public override float DrawChar(DrawingHandleScreen handle, Rune rune, Vector2 ba
                 return metrics.Value.Advance;
             }
 
+            var start = baseline;
             baseline += new Vector2(metrics.Value.BearingX, -metrics.Value.BearingY);
+            if (background_color is Color bg_color)
```
Comment (PJB3005) [inline Robust.Client/Graphics/Font.cs:137]: This is not a good way to implement this. Large spans of background-colored text will break sprite batching and ruin performance.

You should have the text drawing code do separate drawing of the background in one go.

**Review by PJB3005 — CHANGES_REQUESTED** [review]

**Comment by c4llv07e** [comment]
Okay, then I'll think how to do it. Closing for now.

**Pull request closed without merging**